document/core/syntax/modules.rst

```diff
@@ -363,3 +363,9 @@ In each index space, the indices of imports go before the first index of any def
 
 .. note::
    In the current version of WebAssembly, only *immutable* globals may be imported.
+
+   Unlike export names, import names are not necessarily unique.
+   It is possible to import the same |IMODULE|/|INAME| pair multiple times.
+   Even if such imports have different types a module can still be instantiated by :ref:`embedders <embedder>` for which the type difference is irrelevant or that take the expected types into account when resolving imports.
```

Comment:
I find this wording difficult to understand. I think there are two surprising things to highlight: it's surprising that embedders *can* overload on types, since wasm itself can't do that. And it's surprising that embedders can overload on anything *other* than types too, since that's somewhat esoteric.

Reply:
I assume by the second thing you mean being different kinds of entities, like function vs global? Semantically, that still is simply a difference in type, but I changed wording to call out that case explicitly.

Reply:
I also mean like your comment [here](https://github.com/WebAssembly/spec/issues/650#issuecomment-362879847) about resolving imports "positionally". My understanding of core wasm here is that import resolution is just unconstrained; there are no rules about what information embedder may consult to resolve an import. Two imports in a module could be identical in every way, except their position in the import sequence, and still be resolved to different entities, in the same instantiation.

Reply:
Yes, I see. Okay, I dropped the attempt to enumerate possible interpretations/implementations in favour of a more general statement. PTAL